pkg/capture/metrics.go

```diff
@@ -12,108 +12,217 @@ const (
+		},
+			labels,
+		),
+		promBytes: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: config.ServiceName,
+			Subsystem: captureSubsystem,
+			Name:      "bytes_total",
+			Help:      "Number of bytes tracked in flow map",
+		},
+			append(labels, "direction"),
+		),
+		promPackets: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: config.ServiceName,
+			Subsystem: captureSubsystem,
+			Name:      "packets_total",
+			Help:      "Number of packets seen in flow map",
+		},
+			append(labels, "direction"),
+		),
+		promGlobalBufferUsage: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: config.ServiceName,
+			Subsystem: captureSubsystem,
+			Name:      "global_packet_buffer_usage",
+			Help:      "Percentage of global buffer capacity used during flow map rotation",
+		},
+			labels,
+		),
+		promNumFlows: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: config.ServiceName,
+			Subsystem: captureSubsystem,
+			Name:      "flows_total",
+			Help:      "Number of flows tracked in the flow map",
+		},
+			labels,
+		),
+		promPacketsDropped: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: config.ServiceName,
+			Subsystem: captureSubsystem,
+			Name:      "packets_dropped_total",
+			Help:      "Number of packets dropped",
+		},
+			labels,
+		),
+		promCaptureIssues: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: config.ServiceName,
+			Subsystem: captureSubsystem,
+			Name:      "capture_issues_total",
+			Help:      "Number of unexpected issues encountered during packet capture",
+		},
+			append(labels, "issue_type"),
+		),
+		promInterfacesCapturing: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: config.ServiceName,
+			Subsystem: captureManagerSubsystem,
+			Name:      "interfaces_capturing_total",
+			Help:      "Number of interfaces that are actively capturing traffic",
+		}),
+		// not exposing the interface due to the high-cardinality nature of the histogram
+		promRotationDuration: prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: config.ServiceName,
+			Subsystem: captureManagerSubsystem,
+			Name:      "rotation_duration_seconds",
+			Help:      "Total flow map rotation time, aggregated across all interfaces",
+			// rotation is significantly faster than the writeout. Hence the small buckets
+			Buckets: []float64{0.001, 0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 0.75, 1},
+		}),
+	}
+
 	prometheus.MustRegister(
-		promPacketsProcessed,
-		promPacketsDropped,
-		promBytes,
-		promPackets,
-		promGlobalBufferUsage,
-		promNumFlows,
-		promCaptureIssues,
-		promInterfacesCapturing,
-		promRotationDuration,
+		metrics.promPacketsProcessed,
+		metrics.promPacketsDropped,
+		metrics.promBytes,
+		metrics.promPackets,
+		metrics.promGlobalBufferUsage,
+		metrics.promNumFlows,
+		metrics.promCaptureIssues,
+		metrics.promInterfacesCapturing,
+		metrics.promRotationDuration,
 	)
+
+	return metrics
+}
+
```

Comment:
Nit: all of these should be commented with `ObservePackets makes your linter happy....`